From 26b19c18d4df84cebb204d924239aaa745440742 Mon Sep 17 00:00:00 2001
From: zrj <rimvydas.jasinskas@gmail.com>
Date: Sun, 3 Nov 2019 20:22:31 +0200
Subject: [PATCH 4/4] <sys/_types.h>: Move in some types.

 Adjust few types to be more generic.  Add comments.  Fix lib header.
 The __register_t types will be changed separately.
---
 lib/libc/include/libc_private.h |  2 +-
 sys/cpu/x86_64/include/stdint.h | 23 +++--------------------
 sys/sys/_types.h                | 13 +++++++++++++
 3 files changed, 17 insertions(+), 21 deletions(-)

diff --git a/lib/libc/include/libc_private.h b/lib/libc/include/libc_private.h
index c3459e2a57..d634d8a202 100644
--- a/lib/libc/include/libc_private.h
+++ b/lib/libc/include/libc_private.h
@@ -35,7 +35,7 @@
 #ifndef _LIBC_PRIVATE_H_
 #define _LIBC_PRIVATE_H_
 
-#include <machine/types.h>
+#include <sys/_types.h>
 #include <sys/_pthreadtypes.h>
 
 /*
diff --git a/sys/cpu/x86_64/include/stdint.h b/sys/cpu/x86_64/include/stdint.h
index 829c2fe3ec..179e4dc6ac 100644
--- a/sys/cpu/x86_64/include/stdint.h
+++ b/sys/cpu/x86_64/include/stdint.h
@@ -85,15 +85,9 @@ typedef	unsigned int	__bulp_t;
 typedef	__int64_t	__intmax_t;
 typedef	__uint64_t	__uintmax_t;
 
-#ifdef __LP64__
-typedef	__int64_t	__intptr_t;
-typedef	__uint64_t	__uintptr_t;
-typedef	__int64_t	__ptrdiff_t;		/* ptr1 - ptr2 */
-#else
-typedef	__int32_t	__intptr_t;
-typedef	__uint32_t	__uintptr_t;
-typedef	__int32_t	__ptrdiff_t;		/* ptr1 - ptr2 */
-#endif
+typedef	__bslp_t	__intptr_t;
+typedef	__bulp_t	__uintptr_t;
+typedef	__bslp_t	__ptrdiff_t;		/* ptr1 - ptr2 */
 
 typedef	__int32_t	__int_fast8_t;
 typedef	__int32_t	__int_fast16_t;
@@ -120,27 +114,16 @@ typedef	__uint64_t	__uint_least64_t;
  */
 
 #ifdef __LP64__
-typedef __uint64_t	__size_t;
-typedef __int64_t	__ssize_t;
 typedef __int64_t	__register_t;
 typedef __uint64_t	__u_register_t;
 #else
-typedef __uint32_t	__size_t;
-typedef __int32_t	__ssize_t;
 typedef __int32_t	__register_t;
 typedef __uint32_t	__u_register_t;
 #endif
 
-typedef long		__suseconds_t;
-typedef long		__time_t;
-typedef int		__timer_t;
 typedef __int32_t	__sig_atomic_t;	/* XXX */
-typedef unsigned long	__clock_t;
-typedef unsigned long	__clockid_t;
 typedef __uint32_t	__socklen_t;
 typedef volatile int	__atomic_intr_t;
 typedef __int64_t	__rlim_t;
-typedef __int64_t	__off_t;
-typedef __int32_t	__pid_t;
 
 #endif /* _CPU_STDINT_H_ */
diff --git a/sys/sys/_types.h b/sys/sys/_types.h
index 7375b21024..701e7bec44 100644
--- a/sys/sys/_types.h
+++ b/sys/sys/_types.h
@@ -30,4 +30,17 @@
 #include <sys/cdefs.h>
 #include <machine/stdint.h>
 
+/*
+ * Type definitions in various standards.
+ */
+typedef	__int64_t	__clock_t;	/* ticks in CLOCKS_PER_SEC */
+typedef	unsigned long	__clockid_t;	/* CLOCK_* identifiers */
+typedef	__int64_t	__off_t;	/* file offset or size */
+typedef	__int32_t	__pid_t;	/* process [group] id */
+typedef	__bulp_t	__size_t;	/* sizes of objects */
+typedef	__bslp_t	__ssize_t;	/* byte counts or error status */
+typedef	long		__suseconds_t;	/* microseconds (signed) */
+typedef	__int64_t	__time_t;	/* epoch time */
+typedef	int		__timer_t;	/* POSIX timer identifiers */
+
 #endif /* !_SYS__TYPES_H_ */
-- 
2.23.0

